src/commands/compile.ts

```diff
@@ -688,109 +755,240 @@ export async function importLocalFilesToServerSideFolder(wsFolderUri: vscode.Uri
+        return importFileFromContent(doc.name, doc.content, api).then(() => {
+          outputChannel.appendLine("Imported file: " + doc.uri.path.split("/").pop());
+          return doc.name;
+        });
+      })
+    )
+  ).then((results) =>
+    promptForCompile(
+      results.map((result) => (result.status == "fulfilled" ? result.value : null)).filter(notNull),
+      api,
+      true
+    )
+  );
+}
+
+interface XMLQuickPickItem extends vscode.QuickPickItem {
+  file: string;
+}
+
+export async function importXMLFiles(): Promise<any> {
+  try {
+    // Use the server connection from the active document if possible
+    let connectionUri = currentFile()?.uri;
+    if (!connectionUri) {
+      // Use the server connection from a workspace folder
+      const workspaceFolders = vscode.workspace.workspaceFolders || [];
+      if (workspaceFolders.length == 0) {
+        vscode.window.showErrorMessage("'Import XML Files...' command requires an open workspace.", "Dismiss");
+      } else if (workspaceFolders.length == 1) {
+        // Use the current connection
+        connectionUri = workspaceFolders[0].uri;
+      } else {
+        // Pick from the workspace folders
+        connectionUri = (
+          await vscode.window.showWorkspaceFolderPick({
+            ignoreFocusOut: true,
+            placeHolder: "Pick the workspace folder to get server connection information from",
+          })
+        )?.uri;
+      }
+    }
+    if (connectionUri) {
+      const api = new AtelierAPI(connectionUri);
+      // Make sure the server connection is active
+      if (!api.active || api.ns == "") {
+        vscode.window.showErrorMessage(
+          "'Import XML Files...' command requires an active server connection.",
+          "Dismiss"
+        );
+        return;
+      }
+      // Make sure the server has the xml endpoints
+      if (api.config.apiVersion < 7) {
+        vscode.window.showErrorMessage(
+          "'Import XML Files...' command requires InterSystems IRIS version 2023.2 or above.",
+          "Dismiss"
+        );
+        return;
+      }
+      let defaultUri = vscode.workspace.getWorkspaceFolder(connectionUri)?.uri ?? connectionUri;
+      if (defaultUri.scheme != "file") {
+        // Need a default URI with file scheme or the open dialog
+        // will show the virtual files from the workspace folder
+        defaultUri = vscode.workspace.workspaceFile;
+        if (defaultUri.scheme != "file") {
+          vscode.window.showErrorMessage(
+            "'Import XML Files...' command is not supported for unsaved workspaces.",
+            "Dismiss"
+          );
+          return;
+        }
+      }
+      // Prompt the user the file to import
+      let uris = await vscode.window.showOpenDialog({
+        canSelectFiles: true,
+        canSelectFolders: false,
+        canSelectMany: true,
+        openLabel: "Import",
+        filters: {
+          "XML Files": ["xml"],
+        },
+        defaultUri,
```

Comment:
I think we can fix it by dropping the XYZ.code-workspace element from the path. Here's one way:
```suggestion
        defaultUri.with({path: defaultUri.path.split('/').slice(0,-1).join('/')}),
```